src/input/reader_to_storage.cc

```diff
@@ -63,23 +63,25 @@ ReaderToStorage::ReaderToStorage(const FilePath &in_file, const Type::TypeBase &
 	std::ifstream in;
 	in_file.open_stream(in);
 
-    // finish all lazy input types
-    Input::Type::TypeBase::lazy_finish();
+    // finish all input types
+    root_type->finish();
+    Type::TypeBase::delete_unfinished_types();
```

Comment:
Remove deleting here, it is irrelevant for reading process. Allows better control.